graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+    imputer : str
+        The type of missing value imputation to apply to the column.
+        Valid values are "mean", "median" and "most_frequent".
+    """
+
+    def __init__(self, cols: Sequence[str], normalizer: str, imputer: str) -> None:
+        super().__init__(cols)
+        self.cols = cols
+        self.shared_norm = normalizer
+        # Spark uses 'mode' for the most frequent element
+        self.shared_imputation = "mode" if imputer == "most_frequent" else imputer
+
+    def apply(self, input_df: DataFrame) -> DataFrame:
+        logging.debug(
+            "Applying normalizer: %s, imputation: %s", self.shared_norm, self.shared_imputation
+        )
+
+        imputed_df = apply_imputation(self.cols, self.shared_imputation, input_df)
+        scaled_df = apply_norm(self.cols, self.shared_norm, imputed_df)
+
+        # TODO: Figure out why the transformation is producing Double values, and switch to float
+        return scaled_df
+
+    @staticmethod
+    def get_transformation_name() -> str:
+        return "DistNumericalTransformation"
+
+
+class DistMultiNumericalTransformation(DistNumericalTransformation):
+    """Transformation to apply missing value imputation and various forms of normalization
+     to a multi-column numerical input, where the input is a string separated by a delimiter.
+
+    Parameters
+    ----------
+    cols : Sequence[str]
+        The list of columns to apply the transformations on.
+    separator: str
+        The separator string that divides the string values.
+    normalizer : str
+        The normalization to apply to the columns.
+        Valid values are "none", "min-max", and "standard".
+    imputer : str
+        The type of missing value imputation to apply to the column.
+        Valid values are "mean", "median" and "most_frequent".
+    """
+
+    def __init__(
+        self, cols: Sequence[str], separator: Optional[str], normalizer: str, imputer: str
+    ) -> None:
+        assert (
+            len(cols) == 1
+        ), "DistMultiNumericalTransformation only supports one column at a time."
+        super().__init__(cols, normalizer, imputer)
+        self.multi_column = cols[0]
+
+        self.separator = separator
+        # Keep the original separator to split in pure Python
+        self.original_separator = self.separator
+        # Spark's split function uses a regexp so we need to escape
+        # special chars to be used as separators
+        if self.separator in SPECIAL_CHARACTERS:
+            self.separator = f"\\{self.separator}"
+
+    @staticmethod
+    def get_transformation_name() -> str:
+        return "DistMultiNumericalTransformation"
+
+    @staticmethod
+    def apply_norm_vector(vector_col: str, shared_norm: str, vector_df: DataFrame) -> DataFrame:
+        """
+        Applies normalizer column-wise with a single vector column as input.
+        """
+        other_cols = list(set(vector_df.columns).difference([vector_col]))
```

Comment:
Why we need this?

Reply:
We use it to keep track of the other columns in the dataframe to make sure they don't get lost during the call to `select`. On second thought I'm not 100% sure we need this, because each transformer is only supposed to operate on the input columns, the rest are selected away. There might be an edges case I'm forgetting though.

Reply:
Yes verified this is to ensure we have the same behavior as `apply_norm` and `apply_imputation` which will apply the transformations to the columns in `cols` and leave the rest of the columns in the DF unchanged.